File: parte1_framework/src/binarySearch.js
/**
 * Evaluación en Contacto con el Docente - Testing Avanzado
 * UIDE - Diseño de Pruebas, Control de Calidad y Mantenimiento
 * Búsqueda binaria iterativa sobre un arreglo ordenado ascendentemente.
 */

function binarySearch(arr, target) {
  if (!Array.isArray(arr)) {
    throw new TypeError("El primer argumento debe ser un arreglo.");
  }

  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);
    if (arr[mid] === target) {
      return mid;
    } else if (arr[mid] < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return -1;
}

module.exports = { binarySearch };

File: parte2_orquestacion/binary_search.py
"""Algoritmo de busqueda binaria para el pipeline de testing."""


def binary_search(arr, target):
    """Busqueda binaria sobre lista ordenada. Retorna indice o -1."""
    if not isinstance(arr, list):
        raise TypeError("El primer argumento debe ser una lista.")
    left = 0
    right = len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid
        if arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1